wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/urlfactory/DefaultFactoryUrl.java

```diff
@@ -53,8 +46,10 @@ public DefaultFactoryUrl withFactoryFileLocation(String factoryFileLocation) {
   }
 
   @Override
-  public String devfileFileLocation() {
-    return devfileFileLocation;
+  public Map<String, String> devfileFileLocations() {
+    // since for general URL's we didn't have exact filename and always have only one
+    // location, so this kind of map is ok here
+    return singletonMap(null, devfileFileLocation);
```

Comment:
I still have a baaad feeling about this. I'd like to give better comment or suggestion, but I just think it shows some design flaw if you have to do this. It looks very bug-prone and I think it will catch us later with NPE somewhere.

I'd like to see other's opinion. @skabashnyuk  ?

Reply:
This field is used oly once -- it is set as factory source and dashb shows it on factory accepting process.
So basically we can put any appropriate word here instead of filename.

Reply:
I propose to use List<Pair> instead of map.